shared/src/commonTest/kotlin/com/bunbeauty/domain/feature/recommendation/GetRecommendationsUseCaseTest.kt

```diff
@@ -0,0 +1,388 @@
+                maxVisibleCount = 4,
+                recommendationProductList = listOf(
+                    RecommendationProduct(
+                        uuid = "1",
+                        menuProduct = getMenuProduct(
+                            categoryList = listOf(
+                                getCategoryProduct(
+                                    uuid = "1",
+                                )
+                            )
+                        )
+                    ),
+                    RecommendationProduct(
+                        uuid = "2",
+                        menuProduct = getMenuProduct(
+                            categoryList = listOf(
+                                getCategoryProduct(
+                                    uuid = "2",
+                                )
+                            )
+                        )
+                    ),
+                    RecommendationProduct(
+                        uuid = "3",
+                        menuProduct = getMenuProduct(
+                            categoryList = listOf(
+                                getCategoryProduct(
+                                    uuid = "3",
+                                )
+                            )
+                        )
+                    ),
+                    RecommendationProduct(
+                        uuid = "4",
+                        menuProduct = getMenuProduct(
+                            categoryList = listOf(
+                                getCategoryProduct(
+                                    uuid = "4",
+                                )
+                            )
+                        )
+                    )
+                )
+            )
+            coEvery { cartProductInteractor.getCartProductList() } returns listOf(
+                getCartProduct(
+                    uuid = "1",
+                    count = 1,
+                    menuProduct = getMenuProduct(
+                        categoryList = listOf(
+                            getCategoryProduct(
+                                uuid = "1",
+                            )
+                        )
+                    ),
+                )
+            )
+            assertTrue(getRecommendationsUseCase().size == 3)
+        }
+
+    @Test
+    fun `return recommendations when cartProducts has same category but not all and many cartProducts`() =
+        runTest {
+            val categoryList = listOf(
+                getCategoryProduct(
+                    uuid = "1",
+                )
+            )
+            coEvery { recommendationRepository.getRecommendations() } returns getRecommendationProductList(
+                maxVisibleCount = 4,
+                recommendationProductList = listOf(
+                    RecommendationProduct(
+                        uuid = "1",
+                        menuProduct = getMenuProduct(
```

Comment:
1. Lets create `fun getMenuProductWithCategory(categoryUuid = ...)`
2. Lets write it on one line if we only have one param to make it more compact
3. Lets create similar fun for `CartProduct`